### ISSUE TITLE
Exception with filters

### ISSUE THREAD

**Issue by clibu** [appy-one/acebase#14]
Running the following query 
````
db.query('songs')
    .filter('year', '==', 2014)                     // uses the index on year
    .filter('genre', 'in', ['jazz','rock','blues']) // uses the index on genre
    .get();
````
where there is a match on ``filter( year`` but no match on ``filter( genre`` does:
````
Using indexes for query: /songs/*/year, /songs/*/genre
(node:54336) UnhandledPromiseRejectionWarning: Error: TypeError: Cannot read property 'value' of undefined
    at C:\webapps\play2\AceBase\node_modules\acebase-core\dist\data-reference.js:795:19
````
If I change ``filter( genre`` so it does match a doc in 2014 it works as expected.

It doesn't matter whether indexes are used or not.

**Comment by appy-one** [comment]
I was able to reproduce when adding the indexes on year and genre fields. Test code:

```js
const { AceBase } = require('acebase');
const db = new AceBase('test');

async function test() {
    await db.ready();

    // Setup test data
    await db.ref('songs').remove();
    await db.ref('songs').push({ title: 'Some 2014 blues song', year: 2014, genre: 'blues' });
    await db.ref('songs').push({ title: 'Some 2014 rock song', year: 2014, genre: 'rock' });
    await db.ref('songs').push({ title: 'Some 2014 jazz song', year: 2014, genre: 'jazz' });
    await db.ref('songs').push({ title: 'Some 2014 funk song', year: 2014, genre: 'funk' });
    await db.ref('songs').push({ title: 'Some 2014 dance song', year: 2014, genre: 'dance' });

    await db.ref('songs').push({ title: 'Another 2010 blues song', year: 2010, genre: 'blues' });
    await db.ref('songs').push({ title: 'Another 2010 rock song', year: 2010, genre: 'rock' });
    await db.ref('songs').push({ title: 'Another 2010 jazz song', year: 2010, genre: 'jazz' });
    await db.ref('songs').push({ title: 'Another 2010 funk song', year: 2010, genre: 'funk' });
    await db.ref('songs').push({ title: 'Another 2010 dance song', year: 2010, genre: 'dance' });

    // Setup indexes
    await db.indexes.create('songs', 'year');
    await db.indexes.create('songs', 'genre');

    // Query
    const snaps = await db.query('songs')
        .filter('year', '==', 2014)
        .filter('genre', 'in', ['jazz','rock','blues'])
        .get(); // <-- UnhandledPromiseRejectionWarning: Error: TypeError: Cannot read property 'value' of undefined

    console.log(snaps.getValues());
}
test();
```

I've found what's causing this issue and fix it asap

**Comment by appy-one** [comment]
Published fix in version 1.1.1

**Comment by clibu** [comment]
@appy-one Yes fixed. Thanks for the prompt resolution.
